### PR TITLE
CORE-10585 Add E2E test utilities to support member suspension testing

### PR DESCRIPTION
**Pull request by charlieR3**
This adds the e2e test utility functions used in the e2e test repo for testing membership suspension and activation.

This includes a refactor to make utility functions operate in the context of `ClusterInfo` rather than passing the `ClusterInfo` as a parameters. This allows for more readable tests, especially for multi cluster tests.

E2E test repo update: https://github.com/corda/corda-e2e-tests/pull/59

### CODE REVIEW

**Comment by charlieR3** [comment]
Multi cluster support will be added in a follow on PR to reduce complexity for the code reviewers https://github.com/corda/corda-runtime-os/pull/3601

**Comment by ronanbrowne** [comment]
Not sure why BLT is on this one @charlieR3 i only see *.kt changes, better suited to a application dev to review

**Comment by charlieR3** [comment]
> Not sure why BLT is on this one @charlieR3 i only see *.kt changes, better suited to a application dev to review

@ronanbrowne I think it's because I pushed changes and later decided to split them to a PR that will follow this one (https://github.com/corda/corda-runtime-os/pull/3601) so those changes were removed from here.

**Comment by ronanbrowne** [comment]
PR gate "continuous-integration/jenkins/pr-head" is due to a misconfiguration of placeholder job can be ignored, nothing to do with code

**Comment by wzur-r3** [comment]
Hi @charlieR3 - please add relevant reviewers to this PR.

I've removed BLT, as there is nothing to approve from us.

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Member.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Simple data class representing member info returned from a REST API.
+ * Includes functions for parsing commonly accessed data for tests.
+ */
+data class Member(
+    val memberContext: Map<String, String>,
+    val mgmContext: Map<String, String>
+) {
+    val name: String?
+        get() = memberContext["corda.name"]
```
Comment (nikinagy) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Member.kt:15]: can we use the keys from MemberInfoExtension?

Comment (charlieR3) [reply testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Member.kt:15]: I think in general these tests should be decoupled from the application code which is why I've avoided using classes from the runtime-os

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Notary.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package net.corda.e2etest.utilities.types
+
+/**
+ * Simple data class representing a notary service returned from a REST API call.
+ */
```
Comment (nikinagy) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/Notary.kt:10]: Can we add @param info?

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/GroupParameters.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
+package net.corda.e2etest.utilities.types
+
+import net.corda.e2etest.utilities.SimpleResponse
+import net.corda.e2etest.utilities.parseContextMap
+
+/**
+ * Simple data class representing group parameters returned from a REST API.
+ * Includes functions for parsing commonly accessed data for tests.
```
Comment (nikinagy) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/types/GroupParameters.kt:8]: Can we add @param info?

**Review by nikinagy — COMMENTED** [review]
LGTM in general, just two nit comments

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Review by nikinagy — APPROVED** [review]
LGTM

**Pull request merged**